tests/System.Text.Primitives.Tests/Parsing/PrimitiveParserInt16PerfTests.cs

```diff
@@ -0,0 +1,308 @@
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        private static void PrimitiveParserByteSpanToInt16_BytesConsumed_VariableLength_Baseline()
+        {
+            int textLength = s_Int16TextArray.Length;
+            byte[][] utf8ByteArray = (byte[][])Array.CreateInstance(typeof(byte[]), textLength);
+            for (var i = 0; i < textLength; i++)
+            {
+                utf8ByteArray[i] = Text.Encoding.UTF8.GetBytes(s_Int16TextArray[i]);
+            }
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        short.TryParse(s_Int16TextArray[i % textLength], out short value);
+                        TestHelper.DoNotIgnore(value, 0);
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData("๑๐๗๓๗")]
+        [InlineData("๒๑๔๗๔")]
+        [InlineData("๐")]
+        [InlineData("ลบ๒๑๔")]
+        [InlineData("๒๑๔๗๔")]
+        [InlineData("๒")]
+        [InlineData("๒๑๔๗๔")]
+        [InlineData("ลบ๒๑๔")]
+        [InlineData("๒๑๔๗๔")]
+        [InlineData("ลบ๒๑")]
+        [InlineData("ลบ๒")]
+        [InlineData("๒๑๔")]
+        [InlineData("ลบ๒๑๔")]
+        [InlineData("ลบ๒๑๔")]
+        [InlineData("๒๑๔๗")]
+        [InlineData("ลบ๒๑๔")]
+        [InlineData("๒๑๔๗๔")]
+        [InlineData("๒๑")]
+        [InlineData("๒๑๔๗")]
+        [InlineData("ลบ๒๑๔")]
+        [InlineData("+๒๑๔๗๔")]
+        [InlineData("+๒๑")]
+        [InlineData("+๒")]
+        [InlineData("+๒๑๔๗๔")]
+        [InlineData("+๒๘๓๖๔")]
+        [InlineData("+๒๑๔๗")]
+        [InlineData("+๒๑๔๗")]
+        [InlineData("+๒๑๔")]
+        [InlineData("+๒๑๔")]
+        [InlineData("๐๐๐๐๐๐๐๐๐๐๐๐๐๐๐๐๐๐๐๐๑๒๓๕abcdfg")]
+        [InlineData("๒๑๔๗๔abcdefghijklmnop")]
+        [InlineData("๒abcdefghijklmnop")]
+        [InlineData("๒๑๔๗๖abcdefghijklmnop")]
+        [InlineData("ลบ๒๑abcdefghijklmnop")]
+        [InlineData("๒๑๔๗abcdefghijklmnop")]
+        [InlineData("ลบ๒๑abcdefghijklmnop")]
+        [InlineData("ลบ๒abcdefghijklmnop")]
+        [InlineData("๒๑๔abcdefghijklmnop")]
+        [InlineData("ลบ๒๑abcdefghijklmnop")]
+        [InlineData("๒๑abcdefghijklmnop")]
+        [InlineData("+๒๑๔๗๔abcdefghijklmnop")]
+        [InlineData("+๒abcdefghijklmnop")]
+        [InlineData("+๒๑๔๗๖abcdefghijklmnop")]
+        [InlineData("+ลบ๒๑abcdefghijklmnop")]
+        [InlineData("+๒๑๔๗abcdefghijklmnop")]
+        [InlineData("+ลบ๒๑abcdefghijklmnop")]
+        [InlineData("+ลบ๒abcdefghijklmnop")]
+        [InlineData("+๒๑๔abcdefghijklmnop")]
+        [InlineData("+ลบ๒๑abcdefghijklmnop")]
+        [InlineData("+๒๑abcdefghijklmnop")]
+        [InlineData("+ลบ๑๔abcdefghijklmnop")]
+        public unsafe void ParseInt16Thai(string text)
```

Comment:
Is there any particular reason for this to be an unsafe method? If not, I would remove it here and elsewhere.